src/SprykerEco/Zed/Adyen/Business/Hook/Saver/MakePayment/CreditCardSaver.php

```diff
@@ -49,13 +49,23 @@ protected function updatePaymentAdyenTransfer(
      */
     protected function getPaymentStatus(?PaymentAdyenTransfer $paymentAdyenTransfer = null): string
     {
-        if (
-            $paymentAdyenTransfer
-            && $paymentAdyenTransfer->getResultCode() === $this->config->getOmsStatusRefused()
-        ) {
-                return $this->config->getOmsStatusRefused();
+        if ($paymentAdyenTransfer && $this->hasValidResultCode($paymentAdyenTransfer)) {
+            return $paymentAdyenTransfer->getResultCode();
         }
 
-        return $this->config->getOmsStatusAuthorized();
+        return $this->config->getOmsStatusNew();
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentAdyenTransfer $paymentAdyenTransfer
+     *
+     * @return bool
+     */
+    protected function hasValidResultCode(PaymentAdyenTransfer $paymentAdyenTransfer): bool
+    {
+        return in_array($paymentAdyenTransfer->getResultCode(), [
```

Comment:
* Please extract the list of the available statuses to the module config.
* For example it can be `config->getCreditCardPaymentOmsStatusList()`.

Reply:
Done